# Week 11: Introduction to Distributions and Their Use

This notebook answers the questions from week 11 homework

In [14]:
import numpy as np

# Population generation from the prompt
np.random.seed(28)
dBP = np.random.normal(85, 10, 100000)

# Population mean check
population_mean = dBP.mean()
population_mean

np.float64(84.95549142317856)

In [15]:
# Use random seed value of 12 for the tasks below
np.random.seed(12)

# 1) Random sample of 40
sample = np.random.choice(dBP, size=40, replace=False)
sample

array([ 79.51533469,  80.37185499,  86.19968463,  88.66250532,
        88.18490705, 105.38600425,  94.49414083,  88.1509068 ,
        72.51949352,  78.92822907,  91.61642526,  89.29292213,
        87.23535195,  62.89060194,  87.05118312,  71.8427699 ,
        79.36232182,  69.0655521 ,  86.8859776 ,  64.73834414,
        92.72931078,  78.3999444 ,  94.52340652,  79.65875739,
        83.34410933,  78.44953475,  82.5950501 ,  88.12123912,
        97.72985334,  82.70512926,  98.32720723,  93.5128935 ,
        98.22807914,  85.98764706,  99.96454815,  74.21690649,
        92.11131308,  86.74112835,  91.79246656,  75.96782668])

In [16]:
# 2) Sample mean
sample_mean = sample.mean()
sample_mean

np.float64(85.18752155840876)

In [17]:
# 3) Sample standard deviation (sample SD uses ddof=1)
sample_std = sample.std(ddof=1)
sample_std

np.float64(9.627818569777178)

In [18]:
# 4) Standard error
standard_error = sample_std / np.sqrt(len(sample))
standard_error

np.float64(1.5222917789680321)

In [19]:
from scipy import stats

# 5) Confidence coefficient for 95% CI with df = n - 1
c = stats.t.ppf(0.975, df=len(sample) - 1)
c

np.float64(2.022690920036761)

In [20]:
# 6) Lower confidence bound
lower_bound = sample_mean - c * standard_error
lower_bound

np.float64(82.10839579944351)

In [21]:
# 7) Upper confidence bound
upper_bound = sample_mean + c * standard_error
upper_bound

np.float64(88.26664731737401)

In [22]:
# 8) 1000 bootstrapped sample means, sorted ascending
boot_means = np.empty(1000)
for i in range(1000):
    bs = np.random.choice(sample, size=40, replace=True)
    boot_means[i] = bs.mean()
bootstrapped_means = np.sort(boot_means)
len(bootstrapped_means)

1000

In [23]:
# 9) 2.5th percentile
percentile_2_5 = np.percentile(bootstrapped_means, 2.5)
percentile_2_5

np.float64(82.20259484930493)

In [24]:
# 10) 97.5th percentile
percentile_97_5 = np.percentile(bootstrapped_means, 97.5)

results = {
    "sample_mean": sample_mean,
    "sample_std": sample_std,
    "standard_error": standard_error,
    "c": c,
    "lower_bound": lower_bound,
    "upper_bound": upper_bound,
    "percentile_2_5": percentile_2_5,
    "percentile_97_5": percentile_97_5
}
results

{'sample_mean': np.float64(85.18752155840876),
 'sample_std': np.float64(9.627818569777178),
 'standard_error': np.float64(1.5222917789680321),
 'c': np.float64(2.022690920036761),
 'lower_bound': np.float64(82.10839579944351),
 'upper_bound': np.float64(88.26664731737401),
 'percentile_2_5': np.float64(82.20259484930493),
 'percentile_97_5': np.float64(88.1769982523287)}

## Expected Numeric Results

Using the exact setup above:

- `sample_mean` = 85.18752155840876
- `sample_std` = 9.627818569777178
- `standard_error` = 1.5222917789680321
- `c` = 2.022690920036761
- `lower_bound` = 82.10839579944351
- `upper_bound` = 88.26664731737401
- 2.5th percentile of `bootstrapped_means` = 82.20259484930493
- 97.5th percentile of `bootstrapped_means` = 88.1769982523287